packages/veritone-react-common/src/components/SimpleSearchBar/index.js

```diff
@@ -0,0 +1,221 @@
+      focused: false,
+      renderKey: Math.random()
+    });
+  };
+
+  onKeyDown = key => {
+    // handle esc
+    if (this.state.focused && key.keyCode === 27) {
+      this.clear();
+    }
+    // handle enter
+    else if (key.keyCode === 13) {
+      this.props.onSubmit();
+    }
+  };
+
+  render() {
+    const {
+      clickAway,
+      onKeyDown,
+      clear,
+      state: { focused, renderKey },
+      props: {
+        onSubmit,
+        value,
+        autocompleteResults,
+        placeholder,
+        autocomplete,
+        isLoading
+      }
+    } = this;
+
+    const PaperOrDiv = focused ? Paper : Div;
+    const showClear = !!value;
+    const showExtension = !!(
+      autocomplete &&
+      autocompleteResults &&
+      autocompleteResults.length &&
+      value &&
+      focused
+    );
+
+    return (
+      // render key added to force render during cleanup
+      <ClickAwayListener onClickAway={clickAway} key={renderKey}>
+        <PaperOrDiv
+          style={{
+            borderTopLeftRadius: 8,
+            borderTopRightRadius: 8,
+            borderBottomLeftRadius: 8,
+            borderBottomRightRadius: 8
+          }}
+          className={cx(styles.colorContainer)}
+        >
+          <div className={styles.searchBarContainer}>
+            <IconButton disabled={!focused} onClick={onSubmit}>
+              <Search className={styles.iconColor} />
+            </IconButton>
+            <Input
+              // handle esc and enter
+              onKeyDown={onKeyDown}
+              // disable browser auto complete
+              autoComplete="false"
+              // remove material's underline
+              disableUnderline
+              placeholder={placeholder || 'Search'}
+              autoFocus={focused}
+              value={value}
+              onChange={this.onChange}
+              onFocus={this.onFocus}
+              className={cx(styles.input)}
+            />
+            <Zoom in={!!(focused && showClear && isLoading)}>
+              <div className={cx(styles.loadingIcon)}>
+                <CircularProgress size={20} />
+              </div>
+            </Zoom>
+            <Zoom in={showClear} className={cx(styles.clearButton)}>
+              <IconButton onClick={this.clear}>
+                <Clear />
+              </IconButton>
+            </Zoom>
+          </div>
+          {showExtension && (
+            <Fragment>
+              <Divider />
+              <div style={{ overflow: 'scroll' }}>
+                {autocompleteResults.map(result => {
+                  // if we have plain string array
+                  // eslint-disable-next-line lodash/prefer-lodash-typecheck
+                  if (typeof result === 'string') {
+                    return (
+                      <MenuItem
+                        key={result}
+                        // eslint-disable-next-line react/jsx-no-bind
+                        onClick={() => clear() || result.onClick(result)}
```

Comment:
Won't `onClick` here result in an error if `result` is `string`?

Reply:
Result is actually string here. If onClick is null, that would throw error, but we wouldn't care much, would we?